src/WebJobs.Script.Analyzers/AssemblyCache.cs

```diff
@@ -0,0 +1,190 @@
+{
+    // Only support extensions and WebJobs core.
+    // Although extensions may refer to other dlls.
+    public class AssemblyCache
+    {
+        // Map from assembly identities to full paths
+        public static AssemblyCache Instance = new AssemblyCache();
+
+        bool _registered;
+
+        // Assembly Display Name --> Path
+        Dictionary<string, string> _map = new Dictionary<string, string>();
+
+        // Assembly Display Name --> loaded Assembly object
+        Dictionary<string, Assembly> _mapRef = new Dictionary<string, Assembly>();
+
+        const string WebJobsAssemblyName = "Microsoft.Azure.WebJobs";
+        const string WebJobsHostAssemblyName = "Microsoft.Azure.WebJobs.Host";
+
+        JobHostMetadataProvider _tooling;
+
+        internal JobHostMetadataProvider Tooling => _tooling;
+        private int _projectCount;
+
+        // $$$ This can get invoked multiple times concurrently
+        // This will get called on every compilation.
+        // So return early on subsequent initializations.
+        internal void Build(Compilation compilation)
+        {
+            Register();
+
+            int count;
+            lock (this)
+            {
+                // If project references have changed, then reanalyze to pick up new dependencies.
+                var refs = compilation.References.OfType<PortableExecutableReference>().ToArray();
+                count = refs.Length;
+                if ((count == _projectCount) && (_tooling != null))
+                {
+                    return; // already initialized.
+                }
+
+                // Even for netStandard/.core projects, this will still be a flattened list of the full transitive closure of dependencies.
+                foreach (var asm in compilation.References.OfType<PortableExecutableReference>())
+                {
+                    var dispName = asm.Display; // For .net core, the displayname can be the full path
+                    var path = asm.FilePath;
+
+                    _map[dispName] = path;
+                }
+
+                // Builtins
+                _mapRef["mscorlib"] = typeof(object).Assembly;
+                _mapRef[WebJobsAssemblyName] = typeof(Microsoft.Azure.WebJobs.FunctionNameAttribute).Assembly;
+                _mapRef[WebJobsHostAssemblyName] = typeof(Microsoft.Azure.WebJobs.JobHost).Assembly;
+
+                // JSON.Net?
+            }
+
+            // Produce tooling object
+            var webjobsStartups = new List<Type>();
+            foreach (var path in _map.Values)
+            {
+                // We don't want to load and reflect over every dll.
+                // By convention, restrict based on filenames.
+                var filename = Path.GetFileName(path);
+                // TODO: I assume this is so general to avoid missing custom extensions, but can it be tightened up for performance?
+                if (!filename.ToLowerInvariant().Contains("extension"))
```

Comment:
`DurableTask.SqlServer.AzureFunctions.dll` does not have extension in the name. We may have to update this to include durable assemblies

Reply:
Not sure if it is possible to read assemblyNames from extensions.json

Reply:
This logic is looking for `IWebJobsStartup` implementations to pull the relevant binding rules and validation attributes. The Durable Functions bindings are defined in `Microsoft.Azure.WebJobs.Extensions.DurableTask`, so I think skipping `DurableTask.SqlServer.AzureFunctions` would be fine.

If there are binding attributes or `IWebJobsStartup` implementations in an assembly that doesn't follow the `Microsoft.Azure.WebJobs.Extensions.<name>` convention, we'd need to adjust this.

Reply:
You are right, I saw the startup class implementation `DurableTask.SqlServer.AzureFunctions.dll` i assumed it to be implementing IWebJobsStartup , it is implementing 
 FunctionsStartup instead.